assets/javascripts/modules/react/addressChooser.jsx

```diff
@@ -12,55 +14,37 @@ function chooseAddress(addressSection, addressStr) {
     }
 }
 
-export default class ChooseAddress extends React.Component {
-    constructor(props) {
-        super(props)
-        this.state = {
-            value: '-1'
+const augmentAddressOptions = (options) =>
+    (options && options.length > 0) ? [
+        {
+            value: '-1',
+            name: 'Please select an address'
+        },
+        ...options
+    ] : [
+        {
+            value: '-1',
+            name: 'No address found'
         }
-        if(props.options){
-            this.options = [
-                {
-                    value: '-1',
-                    name: 'Please select an address'
-                },
-                ... props.options
-            ]
-        } else {
-            this.options = [
+    ];
 
-                {
-                    value: '-1',
-                    name: 'No address found'
-                }
-            ]
-        }
-        
-    }
 
-    handleChange = (e) => {
-        let target = e.target;
-        this.setState({ value: target.value });
-        chooseAddress(this.props.addressSection, target.value);
-        this.forceUpdate();
-    }
+const createAddressOption = (item, key) => <option key={key} value={item.value}>{item.name}</option>;
+
+const selectAddressOption = (addressSection) => (event) =>  chooseAddress(addressSection, event.target.value);
+
+const ChooseAddress = (props) =>
+    <select className="select select--wide" onChange={selectAddressOption(props.addressSection)} value="-1">
+        {augmentAddressOptions(props.options).map(createAddressOption)}
+    </select>;
+
+export default ChooseAddress;
 
-    render() {
-        let createItem = (item, key) => {
-            return <option key={key} value={item.value}>{item.name}</option>;
-        };
-        return (
-            <select className="select select--wide" onChange={this.handleChange} value={this.state.value}>
-                {this.options.map(createItem)}
```

Comment:
the cause of the bug was that this.options was never updated outside the constructor

Reply:
Has this bug always been there then?

Reply:
I don't know when it was introduced but it doesn't seem to be related to the validation bug we fixed yesterday